python/pyarrow/pandas_compat.py

```diff
@@ -275,11 +275,14 @@ def construct_metadata(columns_to_convert, df, column_names, index_levels,
     else:
         index_descriptors = index_column_metadata = column_indexes = []
 
+    attributes: dict = df.attrs if df.attrs else {}
```

Comment:
Can attrs actually be None? The [NDframe constructor](https://github.com/pandas-dev/pandas/blame/c888af6d0bb674932007623c0867e1fbd4bdc2c6/pandas/core/generic.py#L282) I fround seems to say it'll always be a dict, maybe with values. If this is indeed true (I might be wrong) we could just have:

```suggestion
    attributes = df.attrs
```

If older versions of pandas don't have `attrs` would be safer:
```python
    attributes = df.attrs if hasattr(df, "attrs") else {}
```
Note: type annotations are coming, but we're not sure they'll be inline or not. So I'd suggest not adding it here.

Reply:
Perfect!